discovery/dockerswarm/dockerswarm.go

```diff
@@ -0,0 +1,151 @@
+// Copyright 2020 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dockerswarm
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/docker/docker/api/types/filters"
+	"github.com/docker/docker/client"
+	"github.com/go-kit/kit/log"
+	config_util "github.com/prometheus/common/config"
+	"github.com/prometheus/common/model"
+
+	"github.com/prometheus/prometheus/discovery/refresh"
+	"github.com/prometheus/prometheus/discovery/targetgroup"
+)
+
+const (
+	swarmLabel = model.MetaLabelPrefix + "dockerswarm_"
+)
+
+// DefaultSDConfig is the default Docker Swarm SD configuration.
+var DefaultSDConfig = SDConfig{
+	RefreshInterval: model.Duration(60 * time.Second),
+	Host:            "http://127.0.0.1:2375/",
```

Comment:
Hmm, does it make sense to default this to local? The only other place we do that is Consul, but that's because it's deployed everywhere and you're only meant to talk to the local one.

Reply:
indeed let's make it mandatory.